### PR TITLE
 feat: Provide script to upsert an app setting within an Azure App Service 

### PR DESCRIPTION
**Pull request by pim-simons**
Added a script to upsert an app setting within an Azure App Service.
Based on the great example of @mbraekman https://github.com/mbraekman/Powershell/blob/master/Azure/AppService-UpdateAppSetting.ps1 👍🏻 

Closes https://github.com/arcus-azure/arcus.scripting/issues/236

### CODE REVIEW

File: src/Arcus.Scripting.Tests.Integration/Arcus.Scripting.AppService.tests.ps1 [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+                # Act
+                Set-AzAppServiceSetting -ResourceGroupName $ResourceGroupName -AppServiceName $AppServiceName -AppServiceSettingName $AppServiceSettingName -AppServiceSettingValue $AppServiceSettingValue
+
+                # Assert
+                $actual = Get-AzWebApp -ResourceGroupName $ResourceGroupName -Name $AppServiceName
+                $actual | Should -Not -BeNullOrEmpty 
+
+                $settings = @{ }
+                foreach ($setting in $actual.SiteConfig.AppSettings) 
+                {
```
Comment (stijnmoreels) [inline src/Arcus.Scripting.Tests.Integration/Arcus.Scripting.AppService.tests.ps1:43]:
```suggestion
                foreach ($setting in $actual.SiteConfig.AppSettings) {
```

File: src/Arcus.Scripting.Tests.Integration/Arcus.Scripting.AppService.tests.ps1 [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+                $ResourceGroupName = $config.Arcus.ResourceGroupName
+                $AppServiceName = "unexisting-appservice"
+                $AppServiceSettingName = "somesetting"
+                $AppServiceSettingValue = "somevalue"
+
+                # Act
+                { Set-AzAppServiceSetting -ResourceGroupName $ResourceGroupName -AppServiceName $AppServiceName -AppServiceSettingName $AppServiceSettingName -AppServiceSettingValue $AppServiceSettingValue -ErrorAction Stop } | 
+                    Should -Throw
+            }
+            It "Setting an application setting succeeds" {
```
Comment (stijnmoreels) [inline src/Arcus.Scripting.Tests.Integration/Arcus.Scripting.AppService.tests.ps1:27]: Maybe we should also add a test to set an application setting that already exists? Then, we can maybe more describe this case case, like: "...new setting..."

File: src/Arcus.Scripting.Tests.Integration/appsettings.json [hunk 3]
```diff
@@ -18,6 +18,9 @@
     "IntegrationAccount": {
       "Name": "#{Arcus.IntegrationAccount.Name}#"
     },
+    "AppService": {
+      "Name": "#{Arcus.AppService.Name}#"
```
Comment (stijnmoreels) [inline src/Arcus.Scripting.Tests.Integration/appsettings.json:22]: You found everything on Azure and Azure DevOps? No problems?

File: src/Arcus.Scripting.Tests.Unit/Arcus.Scripting.AppService.tests.ps1 [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+
+                # Act
+                { 
+                   Set-AzAppServiceSetting -ResourceGroupName $ResourceGroupName -AppServiceName $AppServiceName -AppServiceSettingName $AppServiceSettingName -AppServiceSettingValue $AppServiceSettingValue
+                } | Should -Throw -ExpectedMessage "No app service with name '$AppServiceName' could be found in the resource group '$ResourceGroupName'"
+
+                # Assert
+                Assert-VerifiableMock
+            }
+            It "Creating a new setting succeeds" {
```
Comment (stijnmoreels) [inline src/Arcus.Scripting.Tests.Unit/Arcus.Scripting.AppService.tests.ps1:25]: Nit: maybe we can be more descriptive in our test names? Like you did in the first test?

**Review by stijnmoreels — CHANGES_REQUESTED** [review]
Great job! Docs, tests, and implementation. Quality as expected from you. I'm not always active on this repo, so it's a comfort that it still is in good hands with you 😉 .

File: src/Arcus.Scripting.AppService/Arcus.Scripting.AppService.pssproj [hunk 5]
```diff
@@ -0,0 +1,39 @@
+﻿<Project ToolsVersion="4.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <PropertyGroup>
+    <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
+    <SchemaVersion>2.0</SchemaVersion>
+    <ProjectGuid>{342747e9-6406-4da8-b238-37335adba79c}</ProjectGuid>
+    <OutputType>Exe</OutputType>
+    <RootNamespace> Arcus.Scripting.AppService</RootNamespace>
+    <AssemblyName> Arcus.Scripting.AppService</AssemblyName>
+    <Name>Arcus.Scripting.ApiManagement</Name>
```
[the remark this answers is not in the document]
Comment (pim-simons) [reply src/Arcus.Scripting.AppService/Arcus.Scripting.AppService.pssproj:9]: Forgot to change it after copy pasting 😅

File: src/Arcus.Scripting.Tests.Integration/appsettings.json (hunk 3 above)
Comment (pim-simons) [reply src/Arcus.Scripting.Tests.Integration/appsettings.json:22]: Yes no problems, had access to everything I needed 👍🏻

File: src/Arcus.Scripting.Tests.Integration/Arcus.Scripting.AppService.tests.ps1 (hunk 2 above)
Comment (pim-simons) [reply src/Arcus.Scripting.Tests.Integration/Arcus.Scripting.AppService.tests.ps1]: added 2 seperate test cases for create and update

File: src/Arcus.Scripting.AppService/Scripts/Set-AzAppServiceSetting.ps1 [hunk 6]
```diff
@@ -6,60 +6,45 @@ param(
...
-Write-Host "Extracting the existing application settings"
+Write-Host "Azure App service has been found for name '$AppServiceName' in the resource group '$ResourceGroupName'"
+Write-Host "Extracting the existing application settings from Azure App Service"
 $appServiceSettings = $appService.SiteConfig.AppSettings
 
 $existingSettings = @{ }
-Write-Verbose "Existing app settings:"
-foreach ($setting in $appServiceSettings) 
-{
+Write-Verbose "Existing application settings:"
```
Comment (stijnmoreels) [inline src/Arcus.Scripting.AppService/Scripts/Set-AzAppServiceSetting.ps1:21]:
```suggestion
Write-Verbose "Existing Azure App Service application settings:"
```

**Review by stijnmoreels — APPROVED** [review]

File: src/Arcus.Scripting.Tests.Integration/Arcus.Scripting.AppService.tests.ps1 [hunk 7]
```diff
@@ -19,11 +18,46 @@ InModuleScope Arcus.Scripting.AppService {
...
+
+                    $settingsWithoutTestSetting = @{ }
+                    foreach ($setting in $appServiceSettings) {
+                        if ($setting.Name -ne $AppServiceSettingName) {
+                            $settingsWithoutTestSetting[$setting.Name] = $setting.value
+                        }
+                    }
+
+                    Set-AzWebApp -ResourceGroupName $ResourceGroupName -Name $appServiceName -AppSettings $settingsWithoutTestSetting
+                }
```
Comment (stijnmoreels) [inline src/Arcus.Scripting.Tests.Integration/Arcus.Scripting.AppService.tests.ps1:54]: Great job in tear-downing the tests. 🥇

Comment (pim-simons) [reply src/Arcus.Scripting.Tests.Integration/Arcus.Scripting.AppService.tests.ps1:54]: Thanks 👍🏻

**Pull request merged**